src/mlpack/core/cv/k_fold_cv.hpp

```diff
@@ -69,10 +69,14 @@ class KFoldCV
    * @param xs Data points to cross-validate on.
    * @param ys Predictions (labels for classification algorithms and responses
    *     for regression algorithms) for each data point.
+   * @param shuffle Whether or not to shuffle the data and predictions before
+   *     performing cross-validation.  Shuffling will be performed before every
+   *     call to Evaluate().
```

Comment:
When I was implementing `KFoldCV`, I also was thinking about the need to shuffle the passed data. But I was thinking about doing it once during construction of a `KFoldCV` object rather than before each call of `Evaluate()`. Is it a good practice to do it for each set of hyper-parameters? If so, can you provide some references?

Reply:
Agreed, I was not thinking of the hyper-parameter tuning usage.  So I changed the functionality---now if `shuffle` is specified to the constructor, it shuffles during object construction, and shuffling can be performed again manually with the `Shuffle()` function.